### PR TITLE
feat: set default timeout for imap iterator next func

### PR DESCRIPTION
**Pull request by yywing**
https://github.com/celery/billiard/issues/324

### CODE REVIEW

**Comment by yywing** [comment]
https://github.com/celery/billiard/issues/324

**Comment by auvipy** [comment]
can you please explain why it might be a breaking change?

**Comment by yywing** [comment]
I found many timeout not set. I will test it and try to explain why

File: billiard/pool.py [hunk 1]
```diff
@@ -1947,6 +1947,8 @@ def __iter__(self):
         return self
 
     def next(self, timeout=None):
+        if timeout is None:
```
Comment (auvipy) [inline billiard/pool.py:1950]: and if timeout is not None?

**Pull request closed without merging**